### PR TITLE
finish grading

### PR DESCRIPTION
**Pull request by skps2010**

### CODE REVIEW

File: tests/test_course.py [hunk 1]
```diff
@@ -213,3 +213,109 @@ def test_view(self, client_student):
...
+        # modify a score
+        rv = client_admin.put('/course/math/grade/student',
+                              json={
+                                  'title': 'exam2',
+                                  'newTitle': 'exam2 (edit)',
+                                  'content': 'easy',
+                                  'score': 'E',
+                              })
+
+        assert rv.status_code == 200
```
Comment (Bogay) [inline tests/test_course.py]: how about add a check for score data here?

Comment (skps2010) [reply tests/test_course.py]: There's a check at the bottom, which is testing GET

**Comment by Bogay** [comment]
1. need test case for permission check.
   e.g. a student try to edit his/her score, or teachers try to modify other courses' scores
2. some functions may not get excepted behaviour when it comes to multithread?
   for example, if there are two score `hw0`, `hw1` under course `algo` for student `1450`
   first one connection tries to modify `hw1`, at the same time, another connection is deleting `hw0`
   `hw0` may not be deleted if the `course.save()` execute later than other one in `delete_score`

**Comment by Bogay** [comment]
i think it's good enough now.

**Pull request merged**